libmamba/src/core/activation.cpp

```diff
@@ -554,11 +554,11 @@ namespace mamba
         auto conda_environment_env_vars = get_environment_vars(prefix);
 
         // TODO check with conda if that's really what's supposed to happen ...
-        std::remove_if(
+        void(std::remove_if(
```

Comment:
`remove_if` is marked as `[[nodiscard]]` (at least in one of my stdlibs)
There is already `TODO` above to treat this properly, so the easiest way to silence the warning is to use `void(EXPRESSION)`

Reply:
This code is suspicious: either the intent was to move the elements at the end of the container, in which case the elements are not removed and the `push_back` calls that follows will add new eleemnts after the "removed" elements; or it was intended as a remove-erase, in wich case this code doesnt match the intent and hiding the warnings hides the bug.
I'm not sure which one it is here (I'm not familiar with the exact meaning of that code), @JohanMabille or @Hind-M  any idea?

Reply:
Ok now that it's merged I expect the warning to disappear

Reply:
Yes, it disappeared and I merged latest `main` to my branch